backend.php

```diff
@@ -1,49 +1,34 @@
 <?php
 
-use GuzzleHttp\Client;
-use Zipkin\Annotation;
-use Zipkin\Endpoint;
-use Zipkin\Samplers\BinarySampler;
 use Zipkin\Timestamp;
-use Zipkin\TracingBuilder;
+use Zipkin\Propagation\Map;
 
 require_once __DIR__ . '/vendor/autoload.php';
+require_once __DIR__ . '/functions.php';
 
-$endpoint = Endpoint::create('backend', '127.0.0.3', null, 2555);
-$client = new Client();
-
-$logger = new \Monolog\Logger('log');
-$logger->pushHandler(new \Monolog\Handler\ErrorLogHandler());
-
-$reporter = new Zipkin\Reporters\HttpLogging($client, $logger);
-$sampler = BinarySampler::createAsAlwaysSample();
-$tracing = TracingBuilder::create()
-    ->havingLocalEndpoint($endpoint)
-    ->havingSampler($sampler)
-    ->havingReporter($reporter)
-    ->build();
+$tracing = create_tracing('backend', '127.0.0.2');
 
 $request = \Symfony\Component\HttpFoundation\Request::createFromGlobals();
 
 $carrier = array_map(function ($header) {
     return $header[0];
 }, $request->headers->all());
 
-$extractor = $tracing->getPropagation()->getExtractor(new \Zipkin\Propagation\Map());
-$traceContext = $extractor(new ArrayObject($carrier));
+/* Extracts the context from the HTTP headers */
+$extractor = $tracing->getPropagation()->getExtractor(new Map());
+$traceContext = $extractor($carrier);
 
+/* Get users from DB */
 $tracer = $tracing->getTracer();
 $span = $tracer->newChild($traceContext);
```

Comment:
$tracer->newChild($traceContext) can not work with $tracer->newTrace($traceContext) instead, The request to simulate the upstream business is as follows
`
curl -X POST \
  http://xxxxx \
  -H 'cache-control: no-cache' \
  -H 'content-type: application/json' \
  -H 'x-b3-parentspanid: cc1724ac4403a944' \
  -H 'x-b3-sampled: cc1724ac4403a999' \
  -H 'x-b3-spanid: cc1724ac4403a923' \
  -H 'x-b3-traceid: 42438e2a9b566ac7' \
  -d '{
	"user_id": 1,
	"files": [
		"xxxx",
		"12312321"
	]
}'
`

Reply:
@icyxp I don't get the point. Could you please explain it to me?

Reply:
it's ok!